app/services/HttpIdentityService.scala

```diff
@@ -175,13 +188,15 @@ class HttpIdentityService(apiUrl: String, apiClientToken: String)(implicit wsCli
     }
   }
 
-  def getUserIdFromEmail(email: String)(implicit req: RequestHeader, ec: ExecutionContext): EitherT[Future, String, String] = {
+  def getUserIdFromEmail(email: String)(implicit req: RequestHeader, ec: ExecutionContext): EitherT[Future, String, UserIdWithGuestAccountToken] = {
     get(s"user", getHeaders(req), List("emailAddress" -> email)) { resp =>
-      resp.json.validate[UserResponse].asEither.map(_.user.id).leftMap(_.mkString(","))
+      resp.json.validate[UserResponse].asEither.map(
+        userResponse => UserIdWithGuestAccountToken(userResponse.user.id, None)
+      ).leftMap(_.mkString(","))
     }
   }
 
-  def createUserIdFromEmailUser(email: String)(implicit req: RequestHeader, ec: ExecutionContext): EitherT[Future, String, String] = {
+  def createUserIdFromEmailUser(email: String)(implicit req: RequestHeader, ec: ExecutionContext): EitherT[Future, String, UserIdWithGuestAccountToken] = {
```

Comment:
It'd be good at some point if we could use a custom error type rather than a `String` a la the Payment API service you recently implemented in support frontend.